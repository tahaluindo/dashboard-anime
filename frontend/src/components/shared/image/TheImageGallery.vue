<template>
  <section class="flex flex-col h-full gap-3">
    <div class="flex-grow">
      <the-image-content
        class="h-120"
        :src="image"
        :description="description"
      />
    </div>
    <div class="h-20">
      <base-horizontal-slider
        :resizeEvent="optimizedResizeEvent"
        :visibleElements="3"
      >
        <the-image-content
          v-for="(img, index) in imageGallery"
          :key="index"
          :src="img"
          :description="description"
          @click="changeImage(Number(index))"
          class="cursor-pointer"
        />
      </base-horizontal-slider>
    </div>
  </section>
</template>

<script lang="ts">
import { defineComponent, PropType } from "vue";
import { optimizedResizeEvent } from "../../../helpers/optimizedResizeEvent";
import TheImageContent from "../../shared/image/TheImageContent.vue";
import BaseHorizontalSlider from "../slider/BaseHorizontalSlider.vue";

export default defineComponent({
  components: { TheImageContent, BaseHorizontalSlider },
  emits: ["change-image"],
  props: {
    description: {
      type: String,
    },
    image: {
      type: String,
    },
    imageGallery: {
      type: Array as PropType<String[]>,
    },
  },
  setup(_, { emit }) {
    const changeImage = (index: number) => emit("change-image", index);
    return {
      optimizedResizeEvent,
      changeImage,
    };
  },
});
</script>

<style lang="scss"></style>
