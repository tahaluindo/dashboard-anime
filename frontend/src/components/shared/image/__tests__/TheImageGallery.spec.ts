import { mount } from "@vue/test-utils";
import TheImageGallery from "../TheImageGallery.vue";

describe("TheImageGallery test suit", () => {
  it("renders the component with the props passed", async () => {
    const wrapper = mount(TheImageGallery, {
      props: {
        image: "https:/localhost/test",
        imageGallery: ["https:/localhost/test"],
        description: "image",
      },
      shallow: true,
    });

    await wrapper.vm.changeImage(0);
    const emitted: Array<Array<number>> = wrapper.emitted("change-image") || [
      [],
    ];

    expect(emitted[0][0]).toBe(0);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
