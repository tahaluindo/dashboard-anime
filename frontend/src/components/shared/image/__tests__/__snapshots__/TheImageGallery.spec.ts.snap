// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheImageGallery test suit renders the component with the props passed 1`] = `
"<section class=\\"flex flex-col h-full gap-3\\">
  <div class=\\"flex-grow\\">
    <the-image-content-stub class=\\"h-120\\" src=\\"https:/localhost/test\\" description=\\"image\\"></the-image-content-stub>
  </div>
  <div class=\\"h-20\\">
    <base-horizontal-slider-stub resizeevent=\\"[object Object]\\" visibleelements=\\"3\\"></base-horizontal-slider-stub>
  </div>
</section>"
`;
